### PR TITLE
schemachanger: reorganize data-driven integration tests

### LINKED ISSUE #105198: schemachanger: decouple schema change definitions from expected output in data-driven tests
The current situation leads to changes like https://github.com/cockroachdb/cockroach/pull/104864 which appear to be entirely reasonable, however end up having lots of unintended consequences.

What we really want is to separate the data-driven definitions for the schema change from the expected end-to-end test's output. The end-to-end test is only one of many cumulative tests and shouldn't be treated in a special way.

The schema change definition files should include a `setup` and a `test` block and these would be common to all cumulative tests. Each cumulative test should then have its own data-driven test file:
- `TestExecuteWithDMLInjection` should have a file specifying the DML statements to run
- `TestEndToEndSideEffects` should have the expected output using the test dependencies
- `TestRollback` should have the expected EXPLAIN diagrams
- and so forth

This way we can (and should) properly support `skip` directives in the data-driven test files, to only skip one specific test.

Jira issue: CRDB-28920

### PR DESCRIPTION
**Pull request by postamar**
This commit changes how the declarative schema changer's tests are
generated and how the test cases and their expected output are defined.
These are now grouped together in one folder per test case. The expected
output of the side-effects test is separated from the test case
definition itself.

This commit also extends the skip command grammar to require
a comma-separated list of tests to actually skip. This will prevent
wholesale skipping of a particular test case.

This commit also removes the explain_verbose output which just isn't
that useful considering the amount of noise it's accountable for in the
diffs.

Fixes #105198.

Release note: None

### CODE REVIEW

File: pkg/sql/schemachanger/scplan/plan_explain.go [hunk 1]
```diff
@@ -489,10 +489,15 @@ func (p Plan) explainBackfillsAndMerges(root treeprinter.Node, ops []scop.Op) er
...
+				for i, ic := range ics {
 					if ic.IndexID == id {
 						if withComma {
-							sb.WriteString(", ")
+							if i == 0 {
+								sb.WriteRune(';')
+							} else {
+								sb.WriteRune(',')
+							}
+							sb.WriteRune(' ')
```
Comment (postamar) [inline pkg/sql/schemachanger/scplan/plan_explain.go:500]: This is a gratuitous and unrelated improvement to the EXPLAIN (DDL, SHAPE) output. cc @Xiang-Gu

**Review by Xiang-Gu — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105222)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @postamar and @Xiang-Gu)

___
*[`pkg/sql/schemachanger/scplan/plan_explain.go` line 500 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/105222#gh-1235805994:-NYTE36WAUeJKOz8yOH6:b-55px0f) ([raw file](https://github.com/cockroachdb/cockroach/blob/73c1ba20851de92535b0297de3ef50412dfde961/pkg/sql/schemachanger/scplan/plan_explain.go#L500)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

This is a gratuitous and unrelated improvement to the EXPLAIN (DDL, SHAPE) output. cc @Xiang-Gu 
</blockquote></details>

I struggle to understand the code here. What does `;` separate here?

I was hoping we get something like `idx_name(key1, key2; suffix1, suffix2 : stored1, stored2)` where we use two separators for keySuffixCols and storedCols

**Review by rafiss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105222)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @postamar and @Xiang-Gu)

___
*[`pkg/sql/schemachanger/sctest/end_to_end.go` line 196 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/105222#-NYTco-F9b_Xcy60CJ1W:-NYTco-F9b_Xcy60CJ1X:befdwt7) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f748a836f9ba75a30b5cb111c5b8613aab8364d/pkg/sql/schemachanger/sctest/end_to_end.go#L196)):*
> ```Go
> 			d.MaybeScanArgs(t, "tests", &testsCSV)
> 			for _, kind := range strings.Split(testsCSV, ",") {
> 				if kind == "EndToEndSideEffects" {
> ```

could this use `t.Name()` so it's resilient to a test method rename?

**Comment by postamar** [comment]
I forgot about `t.Name()`! Thanks

**Review by Xiang-Gu — APPROVED** [review]
If you don't have a strong feeling of the separator, then I'd really like to have two different ones for keySuffixCols and storedCols.
Other than that, everything LGTM! Thanks for doing this

**Comment by postamar** [comment]
Thanks for the reviews! I put in different separators, not a problem at all.

bors r+

**Pull request merged**